From 71dde40a1a518dace11b7b926d88a63055de7d4c Mon Sep 17 00:00:00 2001
From: David Holsgrove <david.holsgrove@xilinx.com>
Date: Wed, 20 Nov 2013 15:34:53 +0100
Subject: [PATCH 05/28] Add -fstack-usage support

Changelog

2013-03-18  David Holsgrove <david.holsgrove@xilinx.com>

 * gcc/config/microblaze/microblaze.c (microblaze_expand_prologue):
   Add check for flag_stack_usage to enable -fstack-usage support

Signed-off-by: David Holsgrove <david.holsgrove@xilinx.com>
Upstream-Status: Pending
Signed-off-by: Stephan Linz <linz@li-pro.net>
---
 gcc/config/microblaze/microblaze.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gcc/config/microblaze/microblaze.c b/gcc/config/microblaze/microblaze.c
index 7418e49..4417289 100644
--- a/gcc/config/microblaze/microblaze.c
+++ b/gcc/config/microblaze/microblaze.c
@@ -2790,6 +2790,9 @@ microblaze_expand_prologue (void)
 
   fsiz = compute_frame_size (get_frame_size ());
 
+  if (flag_stack_usage)
+    current_function_static_stack_size = fsiz;
+
   /* If this function is a varargs function, store any registers that
      would normally hold arguments ($5 - $10) on the stack.  */
   if (((TYPE_ARG_TYPES (fntype) != 0
-- 
1.8.3.4

